include/deal.II/base/parameter_handler.h

```diff
@@ -1778,6 +1778,37 @@ class ParameterHandler : public Subscriptor
    * The function in essence reads the entire file into a stream and
    * then calls the other parse_input() function with that stream. See
    * there for more information.
+   *
+   * Previous versions of deal.II included a similar function named
+   * <code>read_input</code> which, if the parameter file could not be found
+   * by PathSearch::find, would create a parameter file with default values
+   * and then use that file to set values in the current ParameterHandler. In
+   * order to recapture that behavior one should catch the
+   * PathSearch::ExcFileNotFound exception and then call
+   * ParameterHandler::print_parameters, e.g.,
+   *
+   * @code
+   * try
+   *   {
+   *     parameter_handler.parse_input (filename);
+   *   }
+   * catch (const PathSearch::ExcFileNotFound &)
+   *   {
+   *     std::cerr << "ParameterHandler::read_input: could not open file <"
+   *               << filename
+   *               << "> for reading.\n"
+   *               << "Trying to make file <"
+   *               << filename
+   *               << "> with default values for you."
+   *               << std::endl;
+   *     {
+   *       std::ofstream output (filename);
+   *       parameter_handler.print_parameters (output,
+   *                                           ParameterHandler::OutputStyle::Text);
+   *     }
+   *     parameter_handler.parse_input (filename);
```

Comment:
This line, nor the mention above about setting values to their defaults, is necessary. The value is a parameter is always equal to the default, unless it is explicitly changed by reading an input file.